react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import {
+  FaAngleLeft,
+  FaAngleRight,
+  FaCircle,
+  FaRegCircle,
+} from 'react-icons/fa';
+import { type Image } from './App';
+import './Carousel.css';
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [imageIndex, setImageIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setImageIndex((imageIndex + 1) % images.length);
+    }, 1000);
+    return () => clearTimeout(timerId);
+  }, [setImageIndex, imageIndex, setTimeout, clearTimeout]);
+
+  return (
+    <>
+      <div className="container">
+        <Btn
+          prev="prev"
+          onBtnClick={() =>
+            setImageIndex((imageIndex - 1 + images.length) % images.length)
+          }
+        />
+
+        <ImageDisplay image={images[imageIndex]} />
+
+        <Btn
+          next="next"
+          onBtnClick={() => setImageIndex((imageIndex + 1) % images.length)}
+        />
+      </div>
+
+      <Dots
+        count={images.length}
+        imageIndex={imageIndex}
+        onDotClick={(i) => setImageIndex(i)}
+      />
+    </>
+  );
+}
+
+type BtnProps = {
+  prev?: string;
```

Comment:
If you make these booleans instead of strings, it's a little nicer. You are only using them as booleans in this function, so that code doesn't change, but the JSX that renders the Btn changes a little:

```
<Btn next onClick={} />
<Btn prev onClick={} />
```

In JSX if a prop is an optional boolean prop, if you don't include it in the props it gets the value `undefined` (which is falsy), and if you do include it without a value, it gets the value `true`.